samples/add-map-image-layer/src/main/java/com/esri/arcgismaps/sample/addmapimagelayer/components/AddMapImageLayerViewModel.kt

```diff
@@ -0,0 +1,53 @@
+/* Copyright 2025 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.addmapimagelayer.components
+
+import android.app.Application
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.layers.ArcGISMapImageLayer
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.launch
+
+class AddMapImageLayerViewModel(app: Application) : AndroidViewModel(app) {
+    // The ArcGISMap containing the map image layer
+    val arcGISMap by mutableStateOf(createArcGISMap())
```

Comment:
The map here does not mutate in the sample as it is only set once, also in samples we typically would "build" the ArcGISMap by defining private properties to be consumed by the map: 

```suggestion
    // Create a map image layer using the World Elevation map server
    private val mapImageLayer = ArcGISMapImageLayer(
        url = "https://sampleserver5.arcgisonline.com/arcgis/rest/services/Elevation/WorldElevations/MapServer"
    )

    // The ArcGISMap containing the map image layer
    val arcGISMap = ArcGISMap().apply {
        // Add the map image layer to the operational layers
        operationalLayers.add(mapImageLayer)
    }
```